src/gc/gc.cpp

```diff
@@ -36101,43 +36101,15 @@ CFinalize::FinalizeSegForAppDomain (AppDomain *pDomain,
         // if it has the index we are looking for. If the methodtable is null, it can't be from the
         // unloading domain, so skip it.
         if (method_table(obj) == NULL)
+        {
             continue;
+        }
 
-        // eagerly finalize all objects except those that may be agile.
-        if (obj->GetAppDomainIndex() != pDomain->GetIndex())
+        // does the EE actually want us to finalize this object? (is it agile?)
```

Comment:
>(is it agile?) [](start = 65, length = 14)

I wouldn't even say (is it agile?) here - it's just decision that EE makes for whatever reason it chooses.